### ISSUE TITLE
PHP doesn't get D-Bus signals

### ISSUE THREAD

**Issue by bergie** [midgardproject/midgard-php5#29]
It seems midgard-php doesn't receive D-Bus signals. The following script doesn't output anything (i.e. none of the `notified` callbacks fire):

```
<?php
// Open a connection
$filepath = ini_get("midgard.configuration_file");
$config = new midgard_config();
$config->read_file_at_path($filepath);

function connect($config) {
    $mgd = midgard_connection::get_instance();
    return $mgd->open_config($config);
}

if (!connect($config))
{
    die("Failed to connect\n");
}

// Try session bus
$dbus = new midgard_dbus('/midgard_foo', true);
$dbus->connect('notified', function($dbus) {
    echo "Got notified via session bus\n";
}, array());

$message = "Greetings from midgard-php!(" . mgd_version() . ") PHP ver." . phpversion();
midgard_dbus::send("/midgard_foo", $message, true);

// Try system bus
$dbus = new midgard_dbus('/midgard_foo');
$dbus->connect('notified', function($dbus) {
    echo "Got notified via session bus\n";
}, array());

$message = "Greetings from midgard-php!(" . mgd_version() . ") PHP ver." . phpversion();
midgard_dbus::send("/midgard_foo", $message);

// Try regular signals
$dbus = new midgard_dbus("/mgdschema/midgard_person/update");
$dbus->connect('notified', function($dbus) {
    echo "Got notified via MgdSchema event\n";
}, array());

$person = new midgard_person(1);
$person->firstname = 'Foo';
$person->update();
```

Also I don't see the MgdSchema or the system bus signals with `dbus-monitor --system`, though I do see the session bus signal.

**Comment by bergie** [comment]
This is what I see in Midgard debug log:

```
midgard-core (pid:24674):(debug): midgard_dbus::__construct(...)
midgard-core (pid:24674):(debug):Initialize D-Bus
midgard-core (pid:24674):(debug):New DBus proxy: org.midgardproject.notifier, /org/midgardproject/midgard_foo, org.midgardproject.notifier
midgard-core (pid:24674):(debug): midgard_dbus::connect(...)
midgard-core (pid:24674):(debug): midgard_dbus::send(...)
...etc
```

`dbus-monitor --session` gives the following:

```
signal sender=org.freedesktop.DBus -> dest=(null destination) serial=5 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameOwnerChanged
   string "org.midgardproject.notifier"
   string ""
   string ":1.4960"
method call sender=:1.4960 -> dest=org.freedesktop.DBus serial=8 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=RequestName
   string "org.midgardproject.notifier"
   uint32 0
method call sender=:1.4960 -> dest=org.freedesktop.DBus serial=9 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=GetNameOwner
   string "org.midgardproject.notifier"
method call sender=:1.4960 -> dest=org.midgardproject.notifier serial=10 path=/org/midgardproject/midgard_foo; interface=org.midgardproject.notifier; member=emitNotified
   string "Greetings from midgard-php!(10.05.3) PHP ver.5.3.3-1ubuntu9.3"
signal sender=:1.4960 -> dest=(null destination) serial=11 path=/org/midgardproject/midgard_foo; interface=org.midgardproject.notifier; member=Notified
   string "Greetings from midgard-php!(10.05.3) PHP ver.5.3.3-1ubuntu9.3"
```

PHP `midgard.memory_debug=on` output:

```
MINIT done (pid = 24699)
RINIT
---> g_log_set_handler() => 2
<= RINIT
[0xa919890] php_midgard_gobject_new(midgard_config)
[0xa919890] <= php_midgard_gobject_new()
[0xa919aa8] php_midgard_gobject_new(midgard_connection)
[0xa919aa8] <= php_midgard_gobject_new()
[0xa919a24] midgard_connection::__construct()
---> g_log_remove_handler(..., 2)
---> global_loghandler = 0
[0xa919a24] <= midgard_connection::__construct()
[0xa917720] php_midgard_gobject_new(midgard_dbus)
[0xa917720] <= php_midgard_gobject_new()
[0xa9ddfd0] php_midgard_closure_new_default(zobject = 0xa919b58, zval_array = 0xa917884)
[0xa917be0] php_midgard_gobject_new(midgard_dbus)
[0xa917be0] <= php_midgard_gobject_new()
[0xa917720] __php_midgard_gobject_dtor(midgard_dbus)
[0xa917720] =========> G_IS_OBJECT
[0xa917720] =========> ..._TYPE_NAME != NULL
[0xa917720] =========> gobject's refcount = 2 (before unref)
[0xa917720] __object_properties_dtor()
[0xa917720] <= __object_properties_dtor()
[0xa917720] <= __php_midgard_gobject_dtor()
[0xa9e15f0] php_midgard_closure_new_default(zobject = 0xa9178b0, zval_array = 0xa917b88)
[0xa917e0c] php_midgard_gobject_new(midgard_dbus)
[0xa917e0c] <= php_midgard_gobject_new()
[0xa917be0] __php_midgard_gobject_dtor(midgard_dbus)
[0xa917be0] =========> G_IS_OBJECT
[0xa917be0] =========> ..._TYPE_NAME != NULL
[0xa917be0] =========> gobject's refcount = 2 (before unref)
[0xa917be0] __object_properties_dtor()
[0xa917be0] <= __object_properties_dtor()
[0xa917be0] <= __php_midgard_gobject_dtor()
[0xa9e5558] php_midgard_closure_new_default(zobject = 0xa919b58, zval_array = 0xa917d88)
[0xa918174] php_midgard_gobject_new(midgard_person)
[0xa918174] <= php_midgard_gobject_new()
[0xa9178b0] midgard_person::__construct() 
[0xa6e8b08] php_midgard_object_connect_class_closures(zobject = 0xa9178b0)
[0xa6e8b08] <= php_midgard_object_connect_class_closures(zobject = 0xa9178b0)
[0xa9178b0] <= midgard_person::__construct() 
[0xa9178b0] php_midgard_gobject_write_property()
Done
[0xa918174] __php_midgard_gobject_dtor(midgard_person)
[0xa918174] =========> G_IS_OBJECT
[0xa918174] =========> ..._TYPE_NAME != NULL
[0xa918174] =========> gobject's refcount = 1 (before unref)
[0xa918174] __object_properties_dtor()
[0xa918174] property's [0xa917ebc] refcount: 1 (before) [firstname]
[0xa918174] <= __object_properties_dtor()
[0xa918174] <= __php_midgard_gobject_dtor()
[0xa917e0c] __php_midgard_gobject_dtor(midgard_dbus)
[0xa917e0c] =========> G_IS_OBJECT
[0xa917e0c] =========> ..._TYPE_NAME != NULL
[0xa917e0c] =========> gobject's refcount = 2 (before unref)
[0xa917e0c] __object_properties_dtor()
[0xa917e0c] <= __object_properties_dtor()
[0xa917e0c] <= __php_midgard_gobject_dtor()
[0xa919890] __php_midgard_gobject_dtor(midgard_config)
[0xa919890] =========> G_IS_OBJECT
[0xa919890] =========> ..._TYPE_NAME != NULL
[0xa919890] =========> gobject's refcount = 1 (before unref)
[0xa919890] __object_properties_dtor()
[0xa919890] <= __object_properties_dtor()
[0xa919890] <= __php_midgard_gobject_dtor()
[0xa918e2c] midgard_connection::__destruct()
[0xa918e2c] ---> g_log_remove_handler(..., 3)
[0xa918e2c] ---> (?) global_loghandler != connection's handler
[0xa918e2c] <= midgard_connection::__destruct()
RSHUTDOWN
---> php_midgard_gobject_closure_hash_free()
<= RSHUTDOWN
[0xa919aa8] __php_midgard_gobject_dtor(midgard_connection)
[0xa919aa8] =========> G_IS_OBJECT
[0xa919aa8] =========> ..._TYPE_NAME != NULL
[0xa919aa8] =========> gobject's refcount = 1 (before unref)
[0xa919aa8] __object_properties_dtor()
[0xa919aa8] <= __object_properties_dtor()
[0xa919aa8] <= __php_midgard_gobject_dtor()
```

**Comment by indeyets** [comment]
test for dbus-listener. Closed by b0354e693b12f82eb4e8d1bdaeb2f0e28e0d5f7b
